### ISSUE TITLE
Cannot parse input: expected 'NULL' before: 

### ISSUE THREAD

**Issue by sarancea** [ClickHouse/ClickHouse#31444]
**Can not use dictionaries with "N" in value**

When trying to create a dictionary like:
```
CREATE DICTIONARY default.dict
    (
    `id` UInt32,
    `name` String,
    `sku` String
    )
    PRIMARY KEY id
    SOURCE (HTTP(URL 'http://[::1]/dictionaries/dict.er' CREDENTIALS (USER '***' PASSWORD '***')  FORMAT 'JSONStringsEachRow'))
    LIFETIME (MIN 3600 MAX 5400)
    LAYOUT (HASHED())
```

Response of http://[::1]/dictionaries/dict.er is

`{"id":"1","name":"Name","sku":"NEO2127"}`

` select * from default.dict` returns an error

` Code: 27. DB::Exception: Cannot parse input: expected 'NULL' before: 'ame': (while reading the value of key name): While executing ParallelParsingBlockInputFormat. (CANNOT_PARSE_INPUT_ASSERTION_FAILED) (version 21.11.1.8311 (official build))`

**Comment by den-crane** [comment]
>JSONStringsEachRow
>{"id":"1","name":"Name","sku":"NEO2127"}

It's not JSONStringsEachRow. You should use **JSONEachRow** !!!!

```
{"some_int":42,"some_str":"hello","some_tuple":[1,"a"]} // JSONEachRow

["42","hello","(2,'a')"] // JSONCompactStringsEachRow
```

**Comment by den-crane** [comment]
Though, I am wrong.

**Comment by alexey-milovidov** [comment]
CC @hczhcz

**Comment by alexey-milovidov** [comment]
Fixed in #30497, will be released in 21.12.
